netutils/utils_linux.go

```diff
@@ -62,15 +62,15 @@ func GenerateIfaceName(nlh *netlink.Handle, prefix string, len int) (string, err
 }
 
 // ElectInterfaceAddresses looks for an interface on the OS with the
-// specified name and returns its IPv4 and IPv6 addresses in CIDR
+// specified name and returns its first IPv4 address, all IPv4 addresses,
```

Comment:
I think it is better that this function `"[...] returns all its IPv4 and IPv6 addresses in CIDR notation. If a failure in retrieving the addresses or no IPv4 address is found, an error is returned."`
This will allow us to drop the first *net.IPNet in the return list, which is superflous given we now are adding the v4 list. 
Also, erroring out if no v4 is found (and clearly spelling it out in the API comments), will maintain current logic and facilitate caller's job, which would be able to access the v4List[0] element regardless if the funciton succeeded.

```
v4List, v6List, err := ElectInterfaceAdresses(...)
if err != nil {
    return err
}
nw4 := v4List[0]
If CIDR selector specified {
    for _, nw := range v4List {
        if selector.Contains(nw.IP) {
            nw4 = nw
            break
        }
    }
}

I think @cpuguy83 also suggested something similar while reviewing your docker/docker PR.
```